constraints/miden-vm/range_checker_8bit.air

```diff
@@ -0,0 +1,32 @@
+mod RangeChecker8BitAir
+
+### Helper functions ##############################################################################
+
+# Set to 1 when we don't include the value into the running product.
+fn f0(s0: scalar, s1: scalar) -> scalar:
+    return !s0 & !s1
+
+
+# Set to 1 when we include the value into the running product.
+fn f1(s0: scalar, s1: scalar) -> scalar:
+    return s0 & !s1
+
+
+# Set to 1 when we include two copies of the value into the running product.
+fn f2(s0: scalar, s1: scalar) -> scalar:
+    return !s0 & s1
+
+
+# Set to 1 when we include four copies of the value into the running product.
+fn f3(s0: scalar, s1: scalar) -> scalar:
+    return s0 & s1
```

Comment:
I would probably combine these into a single function which returns a vector with 4 flags. Maybe call it something like `get_multiplicity_flags`.